### DISCUSSION TITLE
柱线组合图，坐标轴转置之后，tooltip显示缺失

### DISCUSSION THREAD

**Discussion by Calmio-Y** [antvis/G2#6184] [Q&A]
### 问题描述

```js
import { Chart } from '@antv/g2';

const data = [
  { time: '10:10', call: 4, waiting: 2, people: 2 },
  { time: '10:15', call: 2, waiting: 6, people: 3 },
  { time: '10:20', call: 13, waiting: 2, people: 5 },
  { time: '10:25', call: 9, waiting: 9, people: 1 },
  { time: '10:30', call: 5, waiting: 2, people: 3 },
  { time: '10:35', call: 8, waiting: 2, people: 1 },
  { time: '10:40', call: 13, waiting: 1, people: 2 },
];

const chart = new Chart({
  container: 'container',
  autoFit: true,
});

chart.data(data);

chart
  .interval()
  .encode('x', 'time')
  .encode('y', 'waiting')
  .axis('y', { title: 'Waiting', titleFill: '#5B8FF9' });

chart
  .line()
  .encode('x', 'time')
  .encode('y', 'people')
  .encode('shape', 'smooth')
  .style('stroke', '#fdae6b')
  .style('lineWidth', 2)
  .scale('y', { independent: true })
  .axis('y', {
    position: 'right',
    grid: null,
    title: 'People',
    titleFill: '#fdae6b',
  });
chart.coordinate({ "transform": [{ "type": "transpose" }] });
chart.render();
```

### 预期行为

期望悬停tooltip显示正常

### 平台

- 操作系统: [macOS, Windows, Linux, React Native ...]
- 网页浏览器: [Google Chrome, Safari, Firefox]

### 屏幕截图或视频（可选）

![image](https://github.com/antvis/G2/assets/30626950/87786987-5310-4715-b27c-c95c5a3bf2b6)

**Comment by pearmini** [comment]
出现这种情况是因为，当鼠标在第一个条的上半区域的时候，没有拾取到折线，所以不会去展示折线的数据。

**Reply by Calmio-Y** [reply]
所以这不算是bug是吗？

**Reply by pearmini** [reply]
不对，应该是一个 bug。我之前解决了水平的情况：https://github.com/antvis/G2/pull/5564 ，但好像 transpose 会有问题，我修复一下。

**Reply by Calmio-Y** [reply]
ok。能麻烦先看看6152吗

**Reply by Calmio-Y** [reply]
能加个急，帮忙看看
![image](https://github.com/antvis/G2/assets/30626950/bdc093f4-6ad2-41b2-8592-ab8343161052)
changeData之后为什么图例没render一下

**Reply by Calmio-Y** [reply]
6191答疑一下下

**Reply by pearmini** [reply]
> changeData之后为什么图例没render一下

这个提供一下完整的复现代码？我不是很清楚具体的问题是啥？

**Reply by Calmio-Y** [reply]
import { Chart } from "@antv/g2";

const chart = new Chart({ container: "container" });
const btn = document.getElementById("btn");

const data1 = [
  { genre: "Sports", sold: 0 },
  { genre: "Strategy", sold: 115 },
  { genre: "Action", sold: 120 },
  { genre: "Shooter", sold: 350 },
  { genre: "Other", sold: 150 },
];

const data2 = [
  { genre: "test", sold: 9 },
  { genre: "name", sold: 4 },
  { genre: "Sports", sold: 6 },
  { genre: "Strategy", sold: 115 },
  { genre: "Action", sold: 120 },
  { genre: "Shooter", sold: 350 },
  { genre: "Other", sold: 150 },
];
chart
  .interval()
  .data(data1)
  .encode("x", "genre")
  .encode("y", "sold")
  .encode("color", "genre")
  .style("minHeight", 50);

chart.render();

btn?.addEventListener("click", () => {
  console.log(888);
  chart.changeData(data2);
});

**Reply by Calmio-Y** [reply]
![image](https://github.com/antvis/G2/assets/30626950/bb7bb18f-68c7-479e-b86d-e00cf2d070cc)
以图片为例。我新的数据第一条genre是name，绘制图例第一个不应该是name吗（如果直接渲染data2，图例的name就在第一个）

**Reply by Calmio-Y** [reply]
有结果吗？

**Reply by pearmini** [reply]
这里采用的策略是增量渲染，尽量不去动之前的渲染位置，目前是符合预期的。

**Reply by Calmio-Y** [reply]
使用方有什么方法去重新渲染位置呢？（非重新初始化一次）

**Reply by pearmini** [reply]
目前好像没有特别好的办法。

**Reply by Calmio-Y** [reply]
在近来的一段时间内，有可能会改增量渲染的逻辑吗？或者增加其他方法去重新渲染

**Reply by pearmini** [reply]
短时间没有这个计划，推荐自己去渲染图例。

**Comment by Calmio-Y** [comment]
柱线组合图，坐标轴转置之后，tooltip显示缺失，这个啥时候修复
